docs/guides/no-suspense.md

```diff
@@ -13,22 +13,20 @@ way you please.
 #### `useStatefulResource.tsx`
 
 ```typescript
-import { useRetrieve, useCache, useError, Schema, ReadShape } from 'rest-hooks';
+import { useRetrieve, useCache, useError, Schema, ReadShape, FetchShape } from 'rest-hooks';
 
 /** If the invalidIfStale option is set we suspend if resource has expired */
-function hasUsableData<
-  S extends Schema,
-  Params extends Readonly<object>,
->(
-  resource: RequestResource<ReadShape<S, Params>> | null,
-  fetchShape: ReadShape<S, Params>,
+export default function hasUsableData(
+  cacheReady: boolean,
+  fetchShape: Pick<FetchShape<any>, 'options'>,
```

Comment:
since this is the only piece taken none of the generics actually have any impact